fdbserver/BlobManifest.actor.cpp

```diff
@@ -443,7 +486,11 @@ class BlobManifestDumper : public ReferenceCounted<BlobManifestDumper> {
 
 	Database db_;
 	Reference<BlobConnectionProvider> blobConn_;
-	Reference<BlobManifestFileSplitter> fileSplitter_;
+	int64_t epoch_;
+	int64_t seqNo_;
+	Reference<AsyncVar<ServerDBInfo> const> dbInfo_;
+	Optional<BlobGranuleCipherKeysCtx> cipherKeysCtx_;
+	bool encryptionEnabled_;
```

Comment:
just curious: if encryption is enabled then `assert(cipherKeysCtx_.present())`  needs to be true; is there a reason to store both?
If yes, then better to add an assert:
`assert(!encryptionEnabled_ || cipherKeysCtx_.present())` in the code. 

Thanks

Reply:
why we have encryptionEnabled_ - encryptionEnabled_ is passed from BlobManager while initializing BlobManifest object. I don't want to check if encryption is enabled every time when dumping manifest(according to previous comment from Josh). so BlobManager checks the encryption and passes it here, so BlobManifest checks it before initializing cipherKeysCtx. seems like a little bit duplication but it can avoid unnecessary encryption checks. 

The assertion is good. let me add it.

Reply:
Can we just rely on `cipherKeysCtx_.present()` to determine if encryption is enabled? There is a duplication, adding assert would make sure both variables don't diverge, however, might be better not to have them unless absolutely needed.

Reply:
seems not easy to remove encryptionEnabled_. I need to construct cipherKeysCtx_  in an actor, but encryptionEnabled_ is passed in the constructor. but I can remove the duplication in another way. I keep encryptionEnabled_ as class variable, but remove cipherKeysCtx_ from the class variable.

Reply:
Makes sense. Yeah I think given `cipherKeysCtx_` is constructed after class initialization, it can be left out as a class variable. 
Thanks